### PR TITLE
Social: Fix attached media disappearing on post publish. 

### PR DESCRIPTION
**Pull request by spsiddarthan**
Fixes 1204116019981263-as-1203876111187914

## Proposed changes:

* When the post gets published, all the connections get toggled off. Due to this, the attached media gets a validation error, due to which the attached media gets removed from the local store. This PR fixes that. 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

1204116019981263-as-1203876111187914
## Does this pull request change what data or activity we track or use?

No
## Testing instructions:

* Without this branch, try publishing a post with an image attached and notice that on publish, when you toggle the social media connection, the image doesn't show up. 
*On this branch, try publishing a post with an image attached and notice that on publish, when you toggle the social media connection, the image will show up. 

Run the test with `Run the tests with:
npm run test -- src/hooks/use-media-restrictions/test/index.test.js in projects/js-packages/publicize-components`

### CODE REVIEW

File: projects/js-packages/publicize-components/src/hooks/use-media-restrictions/index.js [hunk 1]
```diff
@@ -47,6 +47,10 @@ export const getAllowedMediaTypes = enabledConnections => {
 			: DEFAULT_RESTRICTIONS.allowedMediaTypes
 	);
 
+	if ( enabledConnections.length === 0 ) {
+		return DEFAULT_RESTRICTIONS.allowedMediaTypes;
+	}
+
 	if ( typeArrays.length === 0 ) {
 		return [];
```
Comment (pablinos) [inline projects/js-packages/publicize-components/src/hooks/use-media-restrictions/index.js]: Would this change have the same effect?

```suggestion
		return DEFAULT_RESTRICTIONS.allowedMediaTypes;
```

**Review by pablinos — COMMENTED** [review]
I think if I'm reading this right, that we were returning an empty array for allowed media types, which in turn meant that all images were invalid and would be removed?

I'm not sure if the default is the same as 'all media types' but I think we should allow all types when there are no enabled connections.

**Comment by spsiddarthan** [comment]
>I think if I'm reading this right, that we were returning an empty array for allowed media types, which in turn meant that all images were invalid and would be removed?

Correct. Only when the connections are not enabled. 

>I'm not sure if the default is the same as 'all media types' but I think we should allow all types when there are no enabled connections.

The default is all media types. 

>we should allow all types when there are no enabled connections.
Right, I have implemented your suggestion as well, @pablinos

**Review by pablinos — APPROVED** [review]
This works well for me! Thanks for adding the test.

**Pull request merged**